File: 1290. Convert Binary Number in a Linked List to Integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        current=head
        length=0
        while(current!=None):
            length=length+1
            current=current.next
        
        current=head
        number=0
        while(current!=None):
            if(current.val==1):
                number=number+(2 ** (length-1))
            length=length-1
            current=current.next
        return number    

        

File: 1290. Convert Binary Number in a Linked List to Integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        ListNode current=head;
        int length=0;
        while(current!=null)
        {
            length++;
            current=current.next;
        }
        current=head;
        int sum=0;
        while(current!=null)
        {
            if(current.val==1)
                sum+=Math.pow(2,length-1);
            length--;
            current=current.next;
        }
        return sum;
    }
}
